### DISCUSSION TITLE
Question about using InputMixer

### DISCUSSION THREAD

**Discussion by Alfonsol1200** [pschatzmann/arduino-audio-tools#703] [Q&A] [answered]
Hello,
I am trying to mix 2 audio streams from udp and listen them on the audiokit.

It works when i receive data on both udp ports, but if i receive data only on 1 of the udp ports the output is silent. 
is like the mixer is waiting data on both ports to send audio to the output.

for now i am not using a codec to keep it simple. I used as a template the communication_udp_receive example, and i modified it to use the audiokit, and i added a mixer and a second udp port.

any ideas?

**Comment by pschatzmann** [comment, marked as answer]
Yes, the mixer expects data from all defined streams. If you want to stop one stream you need to reconfigure the mixer with end(), add() and begin(). If this is too comberson, feel free to extend the functionality by adding a remove() function...

**Reply by pschatzmann** [reply]
I changed the logic a bit, so if you (e.g. dynamically) update the weight of a channel to 0, it is not read any more
